runtime/src/qio/auxFilesys/hdfs/qio_plugin_hdfs.c

```diff
@@ -147,10 +147,45 @@ qioerr hdfs_preadv (void* file, const struct iovec *vector, int count, off_t off
 
   STARTING_SLOW_SYSCALL;
 
+#ifdef HDFS3
+
+  DO_RETAIN(((hdfs_fs*)fs));
+  
+  hdfs_file* hfl = (hdfs_file*)qio_calloc(sizeof(to_hdfs_file(file)), 1);
+  qio_memcpy(hfl, to_hdfs_file(file), sizeof(to_hdfs_file(file)));
+  qio_memcpy(&(hfl->file), &(to_hdfs_file(file)->file), sizeof(hdfsFile));
+  const int pathnmlen = strlen(to_hdfs_file(file)->pathnm);
+  hfl->pathnm = (char*)qio_calloc(sizeof(char), pathnmlen);
+  qio_memcpy(&(hfl->pathnm), &(to_hdfs_File(file)->pathnm), sizeof(char)*pathnmlen);
+
+  hdfs_fs* hfs = (hdfs_fs*)qio_calloc(sizeof(to_hdfs_fs(fs)), 1);
+  qio_memcpy(hfs, to_hdfs_fs(fs), sizeof(to_hdfs_fs(fs)));
+  qio_memcpy(&(hfs->hfs), &(to_hdfs_fs(fs)->hfs), sizeof(hdfsFS));
+  qio_memcpy(&(hfs->fs_name), &(to_hdfs_fs(fs)->fs_name), sizeof(char)*strlen(to_hdfs_fs(fs)->fs_name));
+
+  hfl->file = hdfsOpenFile(hfs->hfs, hfl->pathnm, O_RDONLY, 0, 0, 0);
```

Comment:
Was attempting to track down the malloc issue - figured dynamically allocating and memcpy'ies would help guide the process - rolled back after some testing and figured out the underlying issue. Everything seems to work well right now for sequential and parallel readers!